lib/downloader.rb

```diff
@@ -0,0 +1,165 @@
+require "rubygems"
+require "bundler/setup"
+require "benchmark"
+require "fileutils"
+require "down/http"
+require "aws-sdk-s3"
+require "logger"
+require "json"
+require "pp"
+require "slack-ruby-client"
+
+class Downloader < LoggingBase
+  attr_reader :frame_url, :base_url
+
+  def initialize(frame_url = "https://v.angelcam.com/iframe?v=9klzdgn2y4")
+    log "making tmp directories"
+    FileUtils.mkdir_p("#{root_dir}/images")
+    FileUtils.mkdir_p("#{root_dir}/videos")
+    @frame_url = frame_url
+  end
+
+  def playlist_url
+    log "finding playlist_url"
+    html = read_url(frame_url)
+    html.match(/(https?:\/\/.*\.m3u8\?token=.*)'/).captures.first
+  end
+
+  def load_ts_segments
+    current_playlist_url = playlist_url
+    @base_url = current_playlist_url.gsub(current_playlist_url.split("/")[-1], "")
+    log "base_url = #{base_url}"
+    item = read_url(current_playlist_url)
+    item.scan(/segment-\d*\.ts/)
+  end
+
+  def download_video(filename)
+    source = base_url + filename
+    log "source = #{source}"
+    destination = "#{root_dir}/videos/#{filename}"
+    log "destination = #{destination}"
+    download_url(source, destination) unless File.exist?(destination)
+    destination
+  end
+
+  private
+
+  def root_dir
+    "/tmp/bigtexan" #NO TRAILING SLASH, PATRICK
```

Comment:
running this locally requires this to exist, correct? should a .gitkeep be thrown in?

Reply:
No, the code creates the folder